sql/flow.go

```diff
@@ -183,3 +190,75 @@ var ExecuteCmdInDocker = func(cmd, mountDirs []string, returnOutput bool) (exitC
 
 	return statusCode, cout, nil
 }
+
+var (
+	ErrNoBaseAstroRuntimeImage = errors.New("base image is not an Astro runtime image in the provided Dockerfile")
+	ErrPythonSDKVersionNotMet  = errors.New("required version for Python SDK dependency not met")
+)
+
+var astroRuntimeVersionRegex = regexp.MustCompile(runtimeImagePrefix + "([^-]*)")
+
+var getAstroDockerfileRuntimeVersion = func() (string, error) {
+	file, err := Os().Open(astroDockerfilePath)
+	if err != nil {
+		return "", err
+	}
+	defer file.Close()
+
+	scanner := BufIo().NewScanner(file)
+	scanner.Scan()
+	text := scanner.Text()
+	if !strings.Contains(text, runtimeImagePrefix) {
+		return "", ErrNoBaseAstroRuntimeImage
+	}
+
+	runtimeVersion := astroRuntimeVersionRegex.FindStringSubmatch(text)[1]
+
+	return runtimeVersion, nil
+}
+
+var EnsurePythonSdkVersionIsMet = func(promptRunner input.PromptRunner) error {
+	astroRuntimeVersion, err := getAstroDockerfileRuntimeVersion()
+	if err != nil {
+		return err
+	}
+	SQLCLIVersion, err := getPypiVersion(astroSQLCliProjectURL)
+	if err != nil {
+		return err
+	}
+	requiredRuntimeVersion, requiredPythonSDKVersion, err := getPythonSDKComptability(astroSQLCliConfigURL, SQLCLIVersion)
+	if err != nil {
+		return err
+	}
+	runtimeVersionMet, err := util.IsRequiredVersionMet(astroRuntimeVersion, requiredRuntimeVersion)
+	if err != nil {
+		return err
+	}
+	if !runtimeVersionMet {
+		result, err := input.PromptGetConfirmation(promptRunner)
+		if err != nil {
+			return err
+		}
+		if !result {
+			return ErrPythonSDKVersionNotMet
+		}
+		requiredPythonSDKDependency := "\nastro-sdk-python" + requiredPythonSDKVersion
```

Comment:
This will be controlled by the compatibility matrix. It may not be always pinned to a specific version. It could also be of the format `>=2.1.0, <3.0.0` which is configured here: https://github.com/astronomer/astro-sdk/blob/astro-cli/sql-cli/config/astro-cli.json